benchmarks/benchmark_guided.py

```diff
@@ -0,0 +1,476 @@
+                    n=n,
+                    temperature=1.0,
+                    top_p=1.0,
+                    ignore_eos=True,
+                    max_tokens=output_len,
+                    guided_decoding=GuidedDecodingParams(
+                        json=SCHEMA) if guided_decoding else None,
+                ))
+
+        generators = []
+        start = time.perf_counter()
+        for i, (prompt, sp) in enumerate(zip(prompts, sampling_params)):
+            generator = llm.generate(prompt, sp, request_id=f"test{i}")
+            generators.append(generator)
+        all_gens = merge_async_iterators(*generators)
+        async for i, res in all_gens:
+            pass
+        end = time.perf_counter()
+        return end - start
+
+
+def main(args: argparse.Namespace):
+    print(args)
+    random.seed(args.seed)
+
+    # Synthesize a prompt with the given input length.
+    tokenizer = AutoTokenizer.from_pretrained(
+        args.tokenizer, trust_remote_code=args.trust_remote_code)
+    prompt = f"Generate an example of a user profile given the following schema: {json.dumps(SCHEMA)}"  # noqa: E501
+    input_len = len(tokenizer(prompt).input_ids)
+    requests = [(prompt, input_len, args.output_len)
+                for _ in range(args.num_prompts)]
+
+    run_args = [
+        requests,
+        args.model,
+        args.tokenizer,
+        args.tensor_parallel_size,
+        args.seed,
+        args.n,
+        args.trust_remote_code,
+        args.dtype,
+        args.max_model_len,
+        args.enforce_eager,
+        args.kv_cache_dtype,
+        args.device,
+        args.enable_prefix_caching,
+        args.enable_chunked_prefill,
+        args.max_num_batched_tokens,
+        args.distributed_executor_backend,
+        args.gpu_memory_utilization,
+        args.num_scheduler_steps,
+        args.disable_async_output_proc,
+        args.guided_decoding,
+        args.warmup,
+    ]
+
+    if args.async_engine:
+        run_args.append(args.disable_frontend_multiprocessing)
+        elapsed_time = uvloop.run(run_vllm_async(*run_args))
+    else:
+        elapsed_time = run_vllm(*run_args)
+    total_num_tokens = sum(prompt_len + output_len
+                           for _, prompt_len, output_len in requests)
+    print(f"Throughput: {len(requests) / elapsed_time:.2f} requests/s, "
+          f"{total_num_tokens / elapsed_time:.2f} tokens/s")
+
+    # Output JSON results if specified
+    if args.output_json:
+        results = {
+            "elapsed_time": elapsed_time,
+            "num_requests": len(requests),
+            "total_num_tokens": total_num_tokens,
+            "requests_per_second": len(requests) / elapsed_time,
+            "tokens_per_second": total_num_tokens / elapsed_time,
+        }
+        with open(args.output_json, "w") as f:
+            json.dump(results, f, indent=4)
+
+
+if __name__ == "__main__":
+    parser = FlexibleArgumentParser(description="Benchmark guided decoding.")
```

Comment:
Could you replace most of these duplicated arguments with the engine's args, like we use in other benchmark scripts?
https://github.com/vllm-project/vllm/blob/ac49b59d8b01ffb9979e18e67b252d45410bc1e6/benchmarks/benchmark_throughput.py#L412-L412